WD-Firmware/comrv/comrv.c

```diff
@@ -18,34 +18,853 @@
+{
+   comrvCacheEntry_t *pCacheEntry;
+   u08_t ucAccumulatedSize = 0, ucIndex = 0;
+   u08_t ucEntryIndex, ucNumberOfCandidates = 0;
+   u32_t uiEvictCandidateMap[D_COMRV_EVICT_CANDIDATE_MAP_SIZE], uiCandidates;
+
+   /* first lets clear the uiCandidates list */
+   comrvMemset(uiEvictCandidateMap, 0, sizeof(u32_t)*D_COMRV_EVICT_CANDIDATE_MAP_SIZE);
+
+#ifdef D_COMRV_EVICTION_LRU
+   /* get the first lru entry */
+   ucEntryIndex = g_stComrvCB.ucLruIndex;
+   /* loop LRU list until the eviction uiCandidates accumulated size
+      reaches the requested eviction size */
+   do
+   {
+      /* point to the cache entry CB */
+      pCacheEntry = &g_stComrvCB.stOverlayCache[ucEntryIndex];
+      /* verify the entry isn't locked */
+      if (pCacheEntry->unProperties.stFields.ucLocked != D_COMRV_ENTRY_LOCKED)
+      {
+         /* count the number of uiCandidates */
+         ucNumberOfCandidates++;
+         /* accumulate size */
+         ucAccumulatedSize += pCacheEntry->unProperties.stFields.ucSizeInMinGroupSizeUnits;
+         /* set the eviction candidate in the eviction map */
+         uiEvictCandidateMap[ucEntryIndex/D_COMRV_DWORD_IN_BITS] |= (1 << ucEntryIndex);
+      }
+      /* move to the next LRU candidate */
+      ucEntryIndex = pCacheEntry->unLru.stFields.typNextLruIndex;
+   /* loop as long as we didn't get to the requested eviction size or we reached end of the list
+      (means that all entries are locked) */
+   } while (ucAccumulatedSize < ucRequestedEvictionSize && ucEntryIndex != D_COMRV_LRU_LAST_ITEM);
+
+#elif defined(D_COMRV_EVICTION_LFU)
+#elif defined(D_COMRV_EVICTION_MIX_LRU_LFU)
+#endif /* D_COMRV_EVICTION_LRU */
+
+   /* now we have eviction uiCandidates bitmap of cache entries - lets create
+      an output sorted list of these entries */
+   for (ucEntryIndex = 0 ; ucIndex != ucNumberOfCandidates && ucEntryIndex < D_COMRV_EVICT_CANDIDATE_MAP_SIZE ; ucEntryIndex++)
+   {
+      /* get the candidates */
+      uiCandidates = uiEvictCandidateMap[ucEntryIndex];
+      /* convert each candidate to an actual value in pEvictCandidatesList */
+      while (uiCandidates)
+      {
+         /* get the lsb that is set */
+         pEvictCandidatesList[ucIndex] = uiCandidates & (-uiCandidates);
+         /* subtract the lsb that is set */
+         uiCandidates -= pEvictCandidatesList[ucIndex]--;
+         pEvictCandidatesList[ucIndex++] += ucEntryIndex*D_COMRV_DWORD_IN_BITS;
+      }
+   }
+
+   /* set the total size of eviction candidates in the last entry */
+   pEvictCandidatesList[ucNumberOfCandidates] = ucAccumulatedSize;
+
+   return ucNumberOfCandidates;
+}
+
+/**
+* search if a specific token is already loaded to the cache
+*
+* @param token - the token to search for
+* @return if the token is loaded the return value is set to the loaded address
+*         otherwise NULL
+*/
+static void* comrvSearchForLoadedOverlayGroup(comrvOverlayToken_t unToken)
```

Comment:
return the index instead of the address and remove the eviction params update

Reply:
done in 2fe7133